src/tribler-core/tribler_core/modules/bandwidth_accounting/community.py

```diff
@@ -75,22 +83,22 @@ def do_payout(self, peer: Peer, amount: int) -> Future:
         with db_session:
             self.database.BandwidthTransaction.insert(tx)
         cache = self.request_cache.add(BandwidthTransactionSignCache(self, tx))
-        self.send_transaction(tx, peer, cache.number)
+        self.send_transaction(tx, peer.address, cache.number)
 
         return cache.future
 
-    def send_transaction(self, transaction: BandwidthTransactionData, peer: Peer, request_id: int) -> None:
+    def send_transaction(self, transaction: BandwidthTransactionData, address: Address, request_id: int) -> None:
         """
         Send a provided transaction to another party.
         :param transaction: The BandwidthTransaction to send to the other party.
-        :param peer: The peer that will receive the transaction.
+        :param peer: The IP address and port of the peer.
         :param request_id: The identifier of the message, is usually provided by a request cache.
         """
         payload = BandwidthTransactionPayload.from_transaction(transaction, request_id)
         packet = self._ez_pack(self._prefix, 1, [payload], False)
-        self.endpoint.send(peer.address, packet)
+        self.endpoint.send(address, packet)
 
-    async def received_transaction(self, source_address: Address, data: bytes) -> None:
+    def received_transaction(self, source_address: Address, data: bytes) -> None:
```

Comment:
Well, I'm a little bit worried that the remote peer can blow up our database by generating an endless stream of fake transactions in response to our query. Do you handle that in some way?

Reply:
Currently no but I acknowledge that this is a concern. I will address this in a subsequent PR, before the release of 7.6 👍